src/Illuminate/Session/Middleware/StartSession.php

```diff
@@ -178,9 +177,8 @@ protected function addCookieToResponse(Response $response, SessionInterface $ses
         }
 
         if ($this->sessionIsPersistent($config = $this->manager->getSessionConfig())) {
-            $response->headers->setCookie(new Cookie(
-                $session->getName(), $session->getId(), $this->getCookieExpirationDate(),
-                $config['path'], $config['domain'], Arr::get($config, 'secure', false)
+            $response->headers->setCookie(cookie()->make(
```

Comment:
i'v also update the helper function cookie(), so we can use it now to take the configuration values .
Also, we can now update without break any previous behavior, as i fix session start and the verify token to have it's previous values if http_only config dose not set.